test/auro-input.test.js

```diff
@@ -212,9 +212,88 @@ describe('auro-input', () => {
 
     await expect(el).to.be.true;
   });
+
+  it('handles credit card formatting - starts with "34" is American Express', async () => {
+    const el = await fixture(html`
+      <auro-input id="format-ccWithIcon" type="credit-card" icon="true" label="Credit Card Number with Icon" required></auro-input>
+    `);
+
+    setInputValue(el, '34');
+    validateCardIcon(el, 'Cc Amex');
+  });
+
+  it('handles credit card formatting - starts with "37" is American Express', async () => {
+    const el = await fixture(html`
+      <auro-input id="format-ccWithIcon" type="credit-card" icon="true" label="Credit Card Number with Icon" required></auro-input>
+    `);
+
+    setInputValue(el, '37');
+    validateCardIcon(el, 'Cc Amex');
+  });
+
+  it('handles credit card formatting - starts with "4" is Visa', async () => {
+    const el = await fixture(html`
+      <auro-input id="format-ccWithIcon" type="credit-card" icon="true" label="Credit Card Number with Icon" required></auro-input>
+    `);
+
+    setInputValue(el, '4');
+    validateCardIcon(el, 'Cc Visa');
+  });
+
+  it('handles credit card formatting - starts with "22" is MasterCard', async () => {
+    const el = await fixture(html`
+      <auro-input id="format-ccWithIcon" type="credit-card" icon="true" label="Credit Card Number with Icon" required></auro-input>
+    `);
+
+    setInputValue(el, '5');
+    validateCardIcon(el, 'Cc Mastercard');
+  });
+
+  it('handles credit card formatting - starts with "644" is Discover Card', async () => {
+    const el = await fixture(html`
+      <auro-input id="format-ccWithIcon" type="credit-card" icon="true" label="Credit Card Number with Icon" required></auro-input>
+    `);
+
+    setInputValue(el, '6');
+    validateCardIcon(el, 'Cc Discover');
+  });
+
+  it('handles credit card formatting - Undefined Value', async () => {
+    const el = await fixture(html`
+      <auro-input id="format-ccWithIcon" type="credit-card" icon="true" label="Credit Card Number with Icon" required></auro-input>
+    `);
+
+    setInputValue(el, undefined);
+    validateCardIcon(el, 'Credit Card');
+  });
+
+  it('handles credit card formatting - Empty Value', async () => {
+    const el = await fixture(html`
+      <auro-input id="format-ccWithIcon" type="credit-card" icon="true" label="Credit Card Number with Icon" required></auro-input>
+    `);
+
+    setInputValue(el, '');
+    validateCardIcon(el, 'Credit Card');
+  });
+
+  it('handles credit card formatting - Alaska Air Visa Cards', async () => {
+    const el = await fixture(html`
+      <auro-input id="format-ccWithIcon" type="credit-card" icon="true" label="Credit Card Number with Icon" required></auro-input>
+    `);
+
+    setInputValue(el, '4147 34');
+    validateCardIcon(el, 'Cc Alaska');
+  });
 });
 
-function setInputValue(input, value) {
+async function validateCardIcon(el, value) {
+  const cardIcon = el.shadowRoot.querySelector('.creditCard-icon');
+  await expect(cardIcon).to.be.accessible();
```

Comment:
We don't need to run this on the icon -- I don't think axe will ever flag an a11y violation here.